antarest/launcher/extensions/adequacy_patch/extension.py

```diff
@@ -30,26 +48,107 @@ def after_export_flat_hook(
         )
-        self._check_config(study_id, study_export_path)
 
-    def _check_config(self, study_id: str, study_export_path: Path) -> None:
-        (
-            study_config,
-            study_tree,
-        ) = self.storage_service.raw_study_service.study_factory.create_from_fs(
+        study = self.study_service.storage_service.raw_study_service.study_factory.create_from_fs(
             study_export_path, study_id, use_cache=False
         )
-        user_config = study_tree.get(["user"])
-        assert_this("flowbased" in user_config or "Flowbased" in user_config)
+        user_config = study.tree.get(["user"])
+        assert_this("flowbased" in user_config)
         adequacy_patch_config = yaml.safe_load(
             cast(
-                bytes, study_tree.get(["user", "adequacypatch", "config.yml"])
+                bytes, study.tree.get(["user", "adequacypatch", "config.yml"])
             )
         )
         assert_this("areas" in adequacy_patch_config)
+        self.prepare_study_for_adq_patch(job_id, study, adequacy_patch_config)
+
+    def prepare_study_for_adq_patch(
+        self, job_id: str, study: FileStudy, adq_patch_config: JSON
+    ) -> Dict[str, bool]:
+        area_to_turn_on: List[str] = [
+            transform_name_to_id(area_id)
+            for area_id in adq_patch_config.get("areas", [])
+        ]
+        original_area_status: Dict[str, bool] = {}
+        original_link_status: Dict[str, bool] = {}
+        for area_id, area in study.config.areas.items():
+            # areas
+            original_area_status[area_id] = "hourly" in area.filters_year
+            if (
+                not original_area_status[area_id]
+                and area_id in area_to_turn_on
+            ):
+                study.tree.save(
+                    ", ".join([*area.filters_year, "hourly"]),
+                    [
+                        "input",
+                        "areas",
+                        area_id,
+                        "optimization",
+                        "filtering",
+                        "filter-year-by-year",
+                    ],
+                )
+
+            # links
+            for area_2, link in area.links.items():
+                link_id = f"{area_id} - {area_2}"
+                original_link_status[link_id] = "hourly" in link.filters_year
+                if not original_link_status[link_id] and (
+                    area_id in area_to_turn_on or area_2 in area_to_turn_on
+                ):
+                    study.tree.save(
+                        ", ".join([*link.filters_year, "hourly"]),
+                        [
+                            "input",
+                            "links",
+                            area_id,
+                            "properties",
+                            area_2,
+                            "filter-year-by-year",
+                        ],
+                    )
+
+        with FileLock(self.tmp_dir / "data.lock"):
```

Comment:
In other files, when FileLock is used, it is used with a constant defined in the class and not directly a string, maybe do the same here ?